sys/net/gnrc/sock/udp/gnrc_sock_udp.c

```diff
@@ -66,15 +65,15 @@ static bool _dyn_port_used(uint16_t port)
 /**
  * @brief   returns a UDP port, and checks for reuse if required
  *
- * complies to RFC 6056, see https://tools.ietf.org/html/rfc6056#section-3.3.3
+ * implements "Another Simple Port Randomization Algorithm" as specified in
+ * RFC 6056, see https://tools.ietf.org/html/rfc6056#section-3.3.2
  */
 static uint16_t _get_dyn_port(sock_udp_t *sock)
 {
     unsigned count = GNRC_SOCK_DYN_PORTRANGE_NUM;
     do {
         uint16_t port = GNRC_SOCK_DYN_PORTRANGE_MIN +
-               (_dyn_port_next * GNRC_SOCK_DYN_PORTRANGE_OFF) % GNRC_SOCK_DYN_PORTRANGE_NUM;
-        _dyn_port_next++;
+               (random_uint32() % GNRC_SOCK_DYN_PORTRANGE_NUM);
         if ((sock == NULL) || (sock->flags & SOCK_FLAGS_REUSE_EP) ||
```

Comment:
Does anybody have an idea why the `sock == NULL` check is necessary?

Reply:
IIRC the compiler complaints that `sock->flags` might not work if `sock` is used uninitialised. I think I was also arguing back then, that this is an internal function and hence can make sure sock is always not `NULL`.

Reply:
But the compiler (or static code checker??) wouldn't calm down without this check.

Reply:
But in that case the check is incorrect as it should be:

```C
         if ((sock != NULL && (sock->flags & SOCK_FLAGS_REUSE_EP)) || 
             !_dyn_port_used(port))
```

then.

Reply:
mhm, you're right the '||' doesn't make sense here. Also I looked up all usages of `_get_dyn_port` and the calling function already makes sure (via `assert` or other) that `sock != NULL`. So I'd say you can safely remove that check, though you could add an `assert(sock)` if you want to be double safe.

Reply:
Ok, good to know. Thanks for researching this. Adjusted the code accordingly.